src/detection_win.cpp

```diff
@@ -550,3 +551,22 @@ void UpdateDevice(PDEV_BROADCAST_DEVICEINTERFACE pDevInf, WPARAM wParam, DeviceS
 
 	SetEvent(deviceChangedRegisteredEvent);
 }
+
+std::string Utf8Encode(const std::string &str)
+{
+    if (str.empty()) {
+        return std::string();
+    }
+
+    //System default code page to wide character
+    int wstr_size = MultiByteToWideChar(CP_ACP, 0, str.c_str(), -1, NULL, 0);
+    std::wstring wstr_tmp(wstr_size, 0);
+    MultiByteToWideChar(CP_ACP, 0, str.c_str(), -1, &wstr_tmp[0], wstr_size);
+
+    //Wide character to Utf8
+    int str_size = WideCharToMultiByte(CP_UTF8, 0, &wstr_tmp[0], (int)wstr_tmp.size(), NULL, 0, NULL, NULL);
+    std::string str_utf8(str_size, 0);
+    WideCharToMultiByte(CP_UTF8, 0, &wstr_tmp[0], (int)wstr_tmp.size(), &str_utf8[0], str_size, NULL, NULL);
+
+    return str_utf8;
+}
```

Comment:
Did you get this code from somewhere like StackOverflow? Let's link it for reference like the following

```cpp
// via https://stackoverflow.com/q/25410284/796832
```

Reply:
No, i wrote it myself, it easy for me, i wrote is before. sometimes i need change the code page from 'gbk' to 'utf8'.

Reply:
There is a bit too much magic here for me to be comfortable. Just seems like something could be exploitable here with all of this reference and byte copying stuff.

Would definitely prefer a standard approach to tackle this

Reply:
Perhaps https://stackoverflow.com/a/3999597/796832